pgmpy/prediction/NaiveBackdoorRegressor.py

```diff
@@ -0,0 +1,284 @@
+        1. X is a pandas DataFrame: It selects the required feature columns based on names.
+        2. X is a NumPy array with `feature_names`: It converts X to a DataFrame
+           and selects the required feature columns based on names.
+        3. X is a NumPy array without `feature_names` (sklearn compatibility case):
+           It assumes the first N columns correspond to the N required features
+           and renames them. Workaround for sklearn's
+           generic test suite, which does not support named features.
+        """
+        required_features = self._extract_feature_columns()
+        X_df = self._ensure_dataframe(X, feature_names)
+
+        # Case 3: Handle sklearn compatibility for generic inputs (e.g., 'feature_0')
+        is_sklearn_generic_input = all(
+            str(col).startswith("feature_") for col in X_df.columns
+        )
+        if is_sklearn_generic_input:
+            if len(X_df.columns) < len(required_features):
+                raise ValueError(
+                    f"Input has {len(X_df.columns)} features, but the causal model "
+                    f"requires {len(required_features)}: {required_features}"
+                )
+            # Select the first N columns and rename them to match the semantic roles.
+            # Workaround for sklearn's check_estimator.
+            feature_df = X_df.iloc[:, : len(required_features)].copy()
+            feature_df.columns = required_features
+            return feature_df
+
+        # Cases 1 & 2: Standard semantic mapping from named columns
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"Required columns based on DAG roles: {required_features}."
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, y, sample_weight: Optional[np.ndarray] = None, feature_names=None):
+        """
+        Fit the Naive Backdoor Regressor.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or pandas DataFrame
+            Training data. If DataFrame, must include columns for exposure,
+            adjustment, and pretreatment variables as defined in causal graph roles.
+            If array, `feature_names` must be provided unless used within a
+            generic sklearn context (like compatibility tests).
+        y : array-like of shape (n_samples,)
+            Target values (outcome variable).
+        sample_weight : array-like of shape (n_samples,), optional
+            Sample weights.
+        feature_names : list, optional
+            Feature names when X is an array. Strongly recommended for array inputs
+            to ensure correct semantic mapping.
+
+        Returns
+        -------
+        self
+            Returns self for method chaining.
+        """
+        X_arr, y_arr = check_X_y(
```

Comment:
It seems to me that using the `validate_data` method is the standard way to check this instead of `check_X_y`.